ocaml/xapi/cert_distrib.ml

```diff
@@ -245,3 +266,37 @@ let go ~__context ~from_hosts ~to_hosts ~existing_cert_strategy =
   List.iter
     (fun host -> Worker.remote_regen_bundle host rpc session_id)
     to_hosts
+
+let collect_pool_certs ~__context ~all_hosts =
+  Helpers.call_api_functions ~__context @@ fun rpc session_id ->
+  all_hosts
+  |> List.map (fun host ->
+         let uuid = Db.Host.get_uuid ~__context ~self:host in
+         let blob = Worker.remote_collect_cert host rpc session_id in
+         (uuid, blob))
+
+let import_joiner ~__context ~uuid ~certificate ~to_hosts =
+  let joiner_certificate = WireProtocol.{uuid; blob= certificate} in
+  Helpers.call_api_functions ~__context @@ fun rpc session_id ->
+  List.iter
+    (fun host ->
+      Worker.remote_write_certs_fs Merge [joiner_certificate] host rpc
+        session_id)
+    to_hosts ;
+  List.iter
+    (fun host -> Worker.remote_regen_bundle host rpc session_id)
+    to_hosts
+
+(* This function is called in the host that is joining a pool *)
+let import_joining_pool_certs ~__context ~pool_certs =
+  let pool_certs =
+    List.map (fun (uuid, blob) -> WireProtocol.{uuid; blob}) pool_certs
+  in
+  Worker.local_write_cert_fs ~__context Merge pool_certs ;
+  Worker.local_regen_bundle ~__context
+
+(* This function is called in the pool that is incorporating a new host *)
```

Comment:
"accepting"?

Reply:
I'd say the pool doesn't have much choice in the matter. ;)